src/codegen.cpp

```diff
@@ -494,6 +495,57 @@ static inline void add_named_global(JuliaVariable *name, void *addr)
     add_named_global(name->name, addr);
 }
 
+static AttributeList get_attrs_ipoeffects(LLVMContext &context, std::pair<bool,uint32_t> effect, bool is_decl)
+{
+    if (!effect.first)
+        return AttributeList();
+    uint32_t effects = effect.second;
+    uint8_t consistent = (effects >> 0) & 0x07;
+    uint8_t effect_free = (effects >> 3) & 0x03;
+    bool nothrow = (effects >> 5) & 0x01;
+    bool terminates = (effects >> 6) & 0x01;
+    bool notaskstate = (effects >> 7) & 0x01;
+    uint8_t inaccessiblememonly = (effects >> 8) & 0x03;
+    bool nonoverlayed = (effects >> 10) & 0x01;
+    bool noinbounds = (effects >> 11) & 0x01;
+    AttrBuilder attr(context);
+
+    if (consistent == 0) {
+        attr.addAttribute("consistent");
+    }
+    if (effect_free == 0) {
+        attr.addAttribute("effect_free");
```

Comment:
Is this not desirable to make readnone/only/write only where applicable

Reply:
Yeah, I think we can add more things to the functions for sure.